File: javascript/LinkedList.js
module.exports = class LinkedList{

    constructor(head = null){
        this.head = head;
    }

    append_node_to_tail = function(new_element){
        let current = this.head;
        if(this.head){
            while(current.next){
                next.previous = current;
                current = current.next;
            }
            current.next = new_element;
        }else{
            this.head = new_element;
        }
    }

    append_node_to_head = function(new_element){
        let current = new_element;
        current.next = this.head;
        this.head = current;
    }

    getTail = function(){
        let current = this.head;
        while(current.next){
            current = current.next;
        }
        return current.value;
    }

    getHead = function(){
        return this.head.value;
    }

    isEmpty = function(){
        if(this.head){
            return true;
        }else{
            return false;
        }
    }
}

File: python/LinkedList.py
class Node():
    def __init__(self, value = None):
        self.value = value
        self.next = None
        self.previous = None

class LinkedList():
    def __init__(self, head = None):
        self.head = head

    def append_node_to_tail(self, new_element):
        current = self.head
        if self.head:
            while current.next:
                current.next.previous = current
                current = current.next                    
            current.next = new_element
        else:
            self.head = new_element
    
    def append_node_to_head(self, new_element):
        current = new_element
        current.next = self.head
        self.head = current
        
    def tail(self):
        current = self.head
        while current.next:
            current = current.next
        return current
    
    def is_empty(self):
        if self.head:
            return False
        else:
            return True